File: src/main/java/zxc/MrDrag0nXYT/nightJoiner/database/DatabaseWorker.java
package zxc.MrDrag0nXYT.nightJoiner.database;

import zxc.MrDrag0nXYT.nightJoiner.database.entity.UserRecord;
import zxc.MrDrag0nXYT.nightJoiner.util.exception.UserRecordNotFound;

import java.sql.Connection;
import java.sql.SQLException;
import java.util.UUID;

public interface DatabaseWorker {
    void initTable(Connection connection) throws SQLException;

    void addUserRecord(Connection connection, UserRecord userRecord) throws SQLException;

    UserRecord getUser(Connection connection, UUID uuid, String username) throws SQLException, UserRecordNotFound;

    String getJoinMessage(Connection connection, UUID uuid, String username) throws SQLException;
    String getQuitMessage(Connection connection, UUID uuid, String username) throws SQLException;

    void setJoinMessage(Connection connection, UUID uuid, String username, String newMessage) throws SQLException;
    void setQuitMessage(Connection connection, UUID uuid, String username, String newMessage) throws SQLException;

    void resetJoinMessage(Connection connection, UUID uuid, String username) throws SQLException;
    void resetQuitMessage(Connection connection, UUID uuid, String username) throws SQLException;
    void resetMessages(Connection connection, UUID uuid, String username) throws SQLException;

    boolean getBlockStatus(Connection connection, String username) throws SQLException;
    boolean getBlockStatus(Connection connection, UUID uuid) throws SQLException;

    void setBlockStatus(Connection connection, String target, int isBlock) throws SQLException, UserRecordNotFound;
}

File: src/main/java/zxc/mrdrag0nxyt/nightJoiner/database/DatabaseWorker.kt
package zxc.mrdrag0nxyt.nightJoiner.database

import zxc.mrdrag0nxyt.nightJoiner.database.entity.UserRecord
import zxc.mrdrag0nxyt.nightJoiner.util.exception.UserRecordNotFound
import java.sql.Connection
import java.sql.SQLException
import java.util.*

interface DatabaseWorker {
    @Throws(SQLException::class)
    fun initTable(connection: Connection)

    @Throws(SQLException::class)
    fun addUserRecord(connection: Connection, userRecord: UserRecord)

    @Throws(SQLException::class, UserRecordNotFound::class)
    fun getUser(connection: Connection, uuid: UUID, username: String?): UserRecord

    @Throws(SQLException::class)
    fun getJoinMessage(connection: Connection, uuid: UUID, username: String?): String?

    @Throws(SQLException::class)
    fun getQuitMessage(connection: Connection, uuid: UUID, username: String?): String?

    @Throws(SQLException::class)
    fun setJoinMessage(connection: Connection, uuid: UUID, username: String, newMessage: String)

    @Throws(SQLException::class)
    fun setQuitMessage(connection: Connection, uuid: UUID, username: String, newMessage: String)

    @Throws(SQLException::class)
    fun resetJoinMessage(connection: Connection, uuid: UUID, username: String)

    @Throws(SQLException::class)
    fun resetQuitMessage(connection: Connection, uuid: UUID, username: String)

    @Throws(SQLException::class)
    fun resetMessages(connection: Connection, uuid: UUID, username: String)

    @Throws(SQLException::class)
    fun getBlockStatus(connection: Connection, username: String?): Boolean

    @Throws(SQLException::class)
    fun getBlockStatus(connection: Connection, uuid: UUID): Boolean

    @Throws(SQLException::class, UserRecordNotFound::class)
    fun setBlockStatus(connection: Connection, target: String?, isBlock: Int)
}
